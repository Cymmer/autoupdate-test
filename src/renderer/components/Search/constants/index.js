export { default as searchTypes } from './searchTypes';
