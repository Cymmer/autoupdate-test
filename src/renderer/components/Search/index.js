import React, { useEffect, useState } from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import { useDebouncedCallback } from 'use-debounce/lib';
import styles from './styles.module.scss';

import {
  ControlledInput,
  ControlledDropdown,
  Text,
  Button,
  IconButton,
  Checkbox,
  Tooltip,
} from '../elements';

import {
  inputTypes,
  dropdownTypes,
  textTypes,
  buttonTypes,
  iconButtonTypes,
} from '../elements/constants';

import searchTypes from './constants/searchTypes';

const Search = ({
  className,
  type,
  filters,
  setFilters,
  filterOptions,
  sortOptions,
  hasSearch,
  isExact,
  onExactChange,
}) => {
  const [areFiltersToggled, toggleFilters] = useState(
    filters[GLOBALS.FILTER_KEYS.FILTER_VALUES]?.length > 0
  );
  const [searchValue, setSearchValue] = useState('');

  useEffect(() => {
    setSearchValue(filters[GLOBALS.FILTER_KEYS.SEARCH_STRING] || '');
  }, []);

  const [onSearch] = useDebouncedCallback((value) => {
    setFilters({
      ...filters,
      [GLOBALS.FILTER_KEYS.SEARCH_STRING]: value,
    });
  }, 500);

  return (
    <div className={cn(styles[`Search___${type}`], className)}>
      <div className={styles.Search_upper}>
        <div className={styles.Search_upper_left}>
          {sortOptions?.length > 0 && (
            <div className={styles.Search_group}>
              <Text
                className={styles.Search_label}
                colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
                type={textTypes.BODY.MD}
              >
                Sort by:
              </Text>
              <ControlledDropdown
                className={styles.Search_upper_dropdown}
                data-test="sortDropdown"
                name="sort"
                options={sortOptions}
                type={dropdownTypes.SLIM}
                value={filters[GLOBALS.FILTER_KEYS.SORT_VALUE]}
                onChange={(val) => {
                  setFilters({
                    ...filters,
                    [GLOBALS.FILTER_KEYS.SORT_VALUE]: val,
                  });
                }}
              />
            </div>
          )}

          {filterOptions?.length > 0 && (
            <Button
              className={styles.Search_button___large}
              data-test="openFiltersButton"
              icon="tune"
              id="filtersButton"
              type={buttonTypes.TEXT.BLUE}
              onClick={() => toggleFilters(!areFiltersToggled)}
            >
              Filters
            </Button>
          )}
        </div>
        <div className={styles.Search_upper_right}>
          {filterOptions?.length > 0 && (
            <IconButton
              className={styles.Search_button___small}
              data-test="closeFiltersButton"
              icon="tune"
              id="filtersButton"
              type={iconButtonTypes.OUTLINE.MD}
              onClick={() => toggleFilters(!areFiltersToggled)}
            >
              Filters
            </IconButton>
          )}
          {hasSearch && (
            <ControlledInput
              className={styles.Search_search}
              data-test="searchFilterInput"
              icon="search"
              name="search"
              placeholder="Search"
              type={inputTypes.SLIM}
              value={searchValue}
              onChange={(e) => {
                const { value } = e.target;
                setSearchValue(value);
                onSearch(value);
              }}
            />
          )}
        </div>
      </div>

      {areFiltersToggled && (
        <div className={styles.Search_lower}>
          <div className={styles.Search_group}>
            <Text
              className={styles.Search_label}
              colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
              type={textTypes.BODY.MD}
            >
              Filter by:
            </Text>
            <ControlledDropdown
              isMulti
              className={styles.Search_lower_dropdown}
              data-test="filtersDropdown"
              name="filters"
              options={filterOptions}
              placeholder="Select filters"
              type={dropdownTypes.SLIM}
              value={filters[GLOBALS.FILTER_KEYS.FILTER_VALUES]}
              onChange={(val) => {
                setFilters({
                  ...filters,
                  [GLOBALS.FILTER_KEYS.FILTER_VALUES]: val || [],
                });
              }}
            />
            {onExactChange != null && (
              <>
                <Tooltip
                  content="If checked, only the results that exactly match the filters will be shown."
                  placement="bottom"
                >
                  <Checkbox
                    checked={isExact === true}
                    className={styles.Search_lower_exact}
                    data-test="exactCheckbox"
                    label="Exact"
                    name="exact"
                    onChange={() => onExactChange(!isExact)}
                  />
                </Tooltip>
              </>
            )}
          </div>
        </div>
      )}
    </div>
  );
};

Search.defaultProps = {
  className: null,
  type: searchTypes.CARD,
  filterOptions: null,
  filters: {},
  sortOptions: null,
  hasSearch: true,
  isExact: false,
  onExactChange: null,
};

Search.propTypes = {
  className: PropTypes.string,
  type: PropTypes.oneOf([searchTypes.BARE, searchTypes.CARD]),

  // The options to be shown in the filters.
  // The values of each of the items in this array
  // matches with the `options` props of our own
  // custom Dropdown component
  filterOptions: PropTypes.oneOfType([
    // Proptype for the non-grouped options
    PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
          .isRequired,
        label: PropTypes.string.isRequired,

        // custom icon in each option (a custom component)
        icon: PropTypes.element,
      })
    ),

    // Proptype for the grouped options
    PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        options: PropTypes.arrayOf(
          PropTypes.shape({
            value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
              .isRequired,
            label: PropTypes.string.isRequired,

            // custom icon in each option (a custom component)
            icon: PropTypes.element,
          })
        ),
      })
    ),
  ]),

  // an object with keys that match the keys
  // in the global `GLOBALS.FILTER_KEYS` which holds all
  // the current filter values
  filters: PropTypes.object,

  // updates the value of the filters
  setFilters: PropTypes.func.isRequired,

  // The options to be shown in the sort.
  // The values of each of the items in this array
  // matches with the `options` props of our own
  // custom Dropdown component
  sortOptions: PropTypes.oneOfType([
    // Proptype for the non-grouped options
    PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
          .isRequired,
        label: PropTypes.string.isRequired,

        // custom icon in each option (a custom component)
        icon: PropTypes.element,
      })
    ),

    // Proptype for the grouped options
    PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        options: PropTypes.arrayOf(
          PropTypes.shape({
            value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
              .isRequired,
            label: PropTypes.string.isRequired,

            // custom icon in each option (a custom component)
            icon: PropTypes.element,
          })
        ),
      })
    ),
  ]),

  // if true, there is a search filter
  hasSearch: PropTypes.bool,

  // these 2 props are for exact filtering
  isExact: PropTypes.bool,
  onExactChange: PropTypes.func,
};

export default Search;
