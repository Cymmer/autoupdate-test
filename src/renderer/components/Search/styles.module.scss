@import '../../scss/variables';
@import '../../scss/mixins';
@import '../../scss/themes';

.Search {
  &___bare {
    margin: 0 0 24px 0;
  }

  &___card {
    padding: 8px 24px;
    border-radius: 16px;
    border: 1px solid;
    @include theme {
      background-color: t('cardBackgroundColor');
      border-color: t('borderColor');
    }
    margin: 0 0 24px 0;
  }

  &_upper {
    @include flex-between;

    @include breakpoint(sm) {
      flex-direction: column-reverse;
    }

    &_left {
      display: flex;
      align-items: center;

      @include breakpoint(sm) {
        width: 100%;
      }
    }

    &_right {
      display: flex;
      align-items: center;
      justify-content: flex-end;

      @include breakpoint(sm) {
        width: 100%;
      }
    }

    &_dropdown {
      max-width: 280px;
      width: 100%;
      margin: 0 32px 0 0;

      @include breakpoint(sm) {
        max-width: 100%;
        margin: 0;
        width: 100%;
      }
    }
  }

  &_lower {
    margin: 16px 0 0 0;

    @include breakpoint(sm) {
      margin: 0;
    }

    &_dropdown {
      width: 100%;
    }

    &_exact {
      margin: 0 4px 0 12px;
    }
  }

  &_group {
    display: flex;
    align-items: center;

    @include breakpoint(sm) {
      margin: 16px 0 0 0;
      width: 100%;
    }
  }

  &_button {
    &___large {
      @include breakpoint(sm) {
        display: none;
      }
    }

    &___small {
      display: none;

      @include breakpoint(sm) {
        display: flex;
        margin: 0 16px 0 0;
        border: none;
      }
    }
  }

  &_search {
    max-width: 280px;
    width: 100%;
    margin: 0 0 0 32px;

    @include breakpoint(sm) {
      max-width: 100%;
      margin: 0;
      width: 100%;
    }
  }

  &_label {
    white-space: nowrap;
    margin: 0 16px 0 0;

    @include breakpoint(xs) {
      white-space: normal;
      margin: 0;
    }
  }
}
