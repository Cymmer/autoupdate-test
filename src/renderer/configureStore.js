import { createHashHistory } from 'history';
import { routerMiddleware } from 'react-router-redux';
import { applyMiddleware, compose, createStore } from 'redux';
import { persistReducer } from 'redux-persist';
import hardSet from 'redux-persist/lib/stateReconciler/hardSet';
import storage from 'redux-persist/lib/storage';
import rootReducer from './ducks/index';

export const configureStore = () => {
  // create middlewares
  const history = createHashHistory();

  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const middleware = applyMiddleware(routerMiddleware(history));

  // create a persist version of the reducer
  const persistConfig = {
    key: 'codechum-desktop',
    storage,
    stateReconciler: hardSet,
  };
  const persistedReducer = persistReducer(persistConfig, rootReducer);

  // create store
  const store = createStore(persistedReducer, composeEnhancers(middleware));

  return store;
};
