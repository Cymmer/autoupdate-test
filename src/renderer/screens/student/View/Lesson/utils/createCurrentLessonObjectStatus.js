import { isLessonObjectActivity } from 'codechum-app-utils';
import getOrCreateLessonObjectStatus from './getOrCreateLessonObjectStatus';

const createCurrentLessonObjectStatus = async ({
  activeLessonObject,
  lessonObjectStatuses,
  nextLessonObject,
  setLessonObjectStatuses,
  studentId,
  university,
}) => {
  if (!activeLessonObject) {
    return;
  }

  const lessonObjectStatus = await getOrCreateLessonObjectStatus({
    studentId,
    lessonObjectId: activeLessonObject.id,
  });
  const newLessonObjectStatuses = { ...lessonObjectStatuses };
  newLessonObjectStatuses[lessonObjectStatus.lesson_object.order] =
    lessonObjectStatus;

  if (
    isLessonObjectActivity(activeLessonObject) &&
    university?.preferences?.has_optional_lesson_activities
  ) {
    const nextLessonObjectStatus = await getOrCreateLessonObjectStatus({
      studentId,
      lessonObjectId: nextLessonObject.id,
    });
    newLessonObjectStatuses[nextLessonObjectStatus.lesson_object.order] =
      lessonObjectStatus;
  }

  setLessonObjectStatuses(newLessonObjectStatuses);
};

export default createCurrentLessonObjectStatus;
