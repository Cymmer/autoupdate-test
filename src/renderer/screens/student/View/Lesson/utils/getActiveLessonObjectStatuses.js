import { LessonObjectStatusesService } from 'services';

const getActiveLessonObjectStatuses = async ({
  activeLesson,
  setIsLoading,
  setLessonObjectStatuses,
  studentId,
}) => {
  setIsLoading(true);
  LessonObjectStatusesService.list({
    isExtended: true,
    params: {
      lessonObjectIds: activeLesson?.lesson_objects
        ?.map((lessonObject) => lessonObject.id)
        .join(','),
      studentId,
    },
  }).then(({ data: { results: retrievedLessonObjectStatuses } }) => {
    setLessonObjectStatuses(
      retrievedLessonObjectStatuses.reduce((acc, curr) => {
        acc[curr.lesson_object.order] = curr;
        return acc;
      }, {})
    );
    setIsLoading(false);
  });
};

export default getActiveLessonObjectStatuses;
