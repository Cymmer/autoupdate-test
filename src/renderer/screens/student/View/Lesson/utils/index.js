export { default as createCurrentLessonObjectStatus } from './createCurrentLessonObjectStatus';
export { default as getActiveLessonObjectStatuses } from './getActiveLessonObjectStatuses';
export { default as getOrCreateLessonObjectStatus } from './getOrCreateLessonObjectStatus';
export { default as setDateTimeFinishedOfLessonObjectStatus } from './setDateTimeFinishedOfLessonObjectStatus';
export { default as updateLessonObjectStatusDateTimeStarted } from './updateLessonObjectStatusDateTimeStarted';
