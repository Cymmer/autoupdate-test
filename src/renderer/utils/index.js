export * from './results';
export * from './testing';
