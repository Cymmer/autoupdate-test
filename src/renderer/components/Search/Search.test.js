import React from 'react';
import { mount, shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { findByTestAttr } from 'codechum-app-utils';
import GLOBALS from 'codechum-app-globals';
import { wrapWithStoreAndRouter } from 'utils';
import Search from './index';
import { configureStore } from '../../configureStore';

jest.useFakeTimers();

describe('RENDERING - Search', () => {
  it('should render with default props', () => {
    const component = shallow(<Search setFilters={() => {}} />);
    expect(component).toMatchSnapshot();
  });

  it('should render the sort options if there is/are sort option/s', () => {
    // test first with an empty array for sortOptions (this should not render
    // the sort options dropdown)
    let component = shallow(<Search setFilters={() => {}} sortOptions={[]} />);
    expect(component).toMatchSnapshot();

    // then test with a non-empty array for sortOptions
    component = shallow(
      <Search
        setFilters={() => {}}
        sortOptions={[
          {
            value: 1,
            label: 'Dummy Label',
          },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the filter options if there is/are filter option/s', () => {
    // test first with an empty array for filterOptions (this should not render
    // the filter options dropdown)
    let component = shallow(
      <Search filterOptions={[]} setFilters={() => {}} />
    );
    expect(component).toMatchSnapshot();

    // then test with a non-empty array for filterOptions
    component = shallow(
      <Search
        filterOptions={[
          {
            value: 1,
            label: 'Dummy Label',
          },
        ]}
        setFilters={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the search filter if it has a search filter', () => {
    const component = shallow(<Search hasSearch setFilters={() => {}} />);
    expect(component).toMatchSnapshot();
  });

  it('should toggle the filters properly', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <Search
          filterOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
          setFilters={() => {}}
        />
      </Provider>
    );
    expect(findByTestAttr(component, 'filtersDropdown').exists()).toEqual(
      false
    );

    const openFiltersButton = findByTestAttr(component, 'openFiltersButton');
    const closeFiltersButton = findByTestAttr(component, 'closeFiltersButton');

    // open the filters
    act(() => {
      openFiltersButton.simulate('click');
    });
    component.update();
    expect(findByTestAttr(component, 'filtersDropdown').exists()).toEqual(true);

    // close the filters
    act(() => {
      closeFiltersButton.simulate('click');
    });
    component.update();
    expect(findByTestAttr(component, 'filtersDropdown').exists()).toEqual(
      false
    );
  });

  it('should not show the exact checkbox if the filters are opened and there is no exact change handler', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <Search
          filterOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
          setFilters={() => {}}
        />
      </Provider>
    );
    expect(findByTestAttr(component, 'exactCheckbox').exists()).toEqual(false);

    const openFiltersButton = findByTestAttr(component, 'openFiltersButton');
    const closeFiltersButton = findByTestAttr(component, 'closeFiltersButton');

    // open the filters
    act(() => {
      openFiltersButton.simulate('click');
    });
    component.update();
    expect(findByTestAttr(component, 'exactCheckbox').exists()).toEqual(false);

    // close the filters
    act(() => {
      closeFiltersButton.simulate('click');
    });
    component.update();
    expect(findByTestAttr(component, 'exactCheckbox').exists()).toEqual(false);
  });

  it('should show the exact checkbox if the filters are opened and there is an exact change handler', () => {
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <Search
          filterOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
          setFilters={() => {}}
          onExactChange={() => {}}
        />
      </Provider>
    );
    expect(findByTestAttr(component, 'exactCheckbox').exists()).toEqual(false);

    const openFiltersButton = findByTestAttr(component, 'openFiltersButton');
    const closeFiltersButton = findByTestAttr(component, 'closeFiltersButton');

    // open the filters
    act(() => {
      openFiltersButton.simulate('click');
    });
    component.update();
    expect(findByTestAttr(component, 'exactCheckbox').exists()).toEqual(true);

    // close the filters
    act(() => {
      closeFiltersButton.simulate('click');
    });
    component.update();
    expect(findByTestAttr(component, 'exactCheckbox').exists()).toEqual(false);
  });
});

describe('EVENTS - Search', () => {
  it('should be able to change the sorting', () => {
    const existingFilters = {
      [GLOBALS.FILTER_KEYS.SEARCH_STRING]: 'Dummy Search String',
    };

    const setFiltersFn = jest.fn();
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <Search
          filters={existingFilters}
          setFilters={setFiltersFn}
          sortOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
        />
      </Provider>
    );

    const newValue = {
      value: 2,
      label: 'New Dummy Label',
    };

    findByTestAttr(component, 'sortDropdown').first().prop('onChange')(
      newValue
    );

    expect(setFiltersFn).toHaveBeenCalledWith({
      ...existingFilters,
      [GLOBALS.FILTER_KEYS.SORT_VALUE]: newValue,
    });
  });

  it('should be able to change the search string', async () => {
    const existingFilters = {
      [GLOBALS.FILTER_KEYS.SORT_VALUE]: {
        value: 1,
        label: 'Dummy Label',
      },
    };

    const setFiltersFn = jest.fn();
    const component = mount(
      wrapWithStoreAndRouter(
        <Search
          hasSearch
          filters={existingFilters}
          setFilters={setFiltersFn}
          sortOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
        />
      )
    );

    const newSearchString = 'Dummy Search String';

    await act(async () => {
      await findByTestAttr(component, 'searchFilterInput').prop('onChange')({
        target: {
          value: newSearchString,
        },
      });

      // advance the timer by 500 to immediately finish the 500ms waiting
      // time for the debounce function of the service call
      jest.advanceTimersByTime(500);
    });

    expect(setFiltersFn).toHaveBeenCalledWith({
      ...existingFilters,
      [GLOBALS.FILTER_KEYS.SEARCH_STRING]: newSearchString,
    });
  });

  it('should be able to change the filters', () => {
    const existingFilters = {
      [GLOBALS.FILTER_KEYS.SORT_VALUE]: {
        value: 1,
        label: 'Dummy Label',
      },
    };

    const setFiltersFn = jest.fn();
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <Search
          hasSearch
          filterOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
          filters={existingFilters}
          setFilters={setFiltersFn}
          sortOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
        />
      </Provider>
    );

    // open the filters
    act(() => {
      findByTestAttr(component, 'openFiltersButton').simulate('click');
    });
    component.update();

    const newFilterValue = {
      value: 1,
      label: 'Dummy Label',
    };

    findByTestAttr(component, 'filtersDropdown').first().prop('onChange')(
      newFilterValue
    );

    expect(setFiltersFn).toHaveBeenCalledWith({
      ...existingFilters,
      [GLOBALS.FILTER_KEYS.FILTER_VALUES]: newFilterValue,
    });
  });

  it('should be able to change the exact checkbox', () => {
    const existingFilters = {
      [GLOBALS.FILTER_KEYS.SORT_VALUE]: {
        value: 1,
        label: 'Dummy Label',
      },
    };

    const setFiltersFn = jest.fn();
    const exactChangeFn = jest.fn();
    const store = configureStore();
    const component = mount(
      <Provider store={store}>
        <Search
          hasSearch
          filterOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
          filters={existingFilters}
          isExact={false}
          setFilters={setFiltersFn}
          sortOptions={[
            {
              value: 1,
              label: 'Dummy Label',
            },
          ]}
          onExactChange={exactChangeFn}
        />
      </Provider>
    );

    // open the filters
    act(() => {
      findByTestAttr(component, 'openFiltersButton').simulate('click');
    });
    component.update();

    findByTestAttr(component, 'exactCheckbox').prop('onChange')();
    expect(exactChangeFn).toHaveBeenCalledWith(true);
  });
});
