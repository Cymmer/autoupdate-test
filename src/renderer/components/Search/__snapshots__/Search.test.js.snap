// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Search should render the filter options if there is/are filter option/s 1`] = `
<div
  className="Search___card"
>
  <div
    className="Search_upper"
  >
    <div
      className="Search_upper_left"
    />
    <div
      className="Search_upper_right"
    >
      <ControlledInput
        autoComplete={null}
        className="Search_search"
        data-test="searchFilterInput"
        disabled={false}
        error={null}
        helperText={null}
        icon="search"
        id={null}
        kind="text"
        label={null}
        max={null}
        maxLength={null}
        min={null}
        name="search"
        onChange={[Function]}
        placeholder="Search"
        step={null}
        success={null}
        tabIndex={null}
        type="slim"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`RENDERING - Search should render the filter options if there is/are filter option/s 2`] = `
<div
  className="Search___card"
>
  <div
    className="Search_upper"
  >
    <div
      className="Search_upper_left"
    >
      <Button
        className="Search_button___large"
        data-test="openFiltersButton"
        dataTestId={null}
        disabled={false}
        icon="tune"
        id="filtersButton"
        isLoading={false}
        isLocked={false}
        kind="button"
        lockedIconClassName={null}
        onClick={[Function]}
        tabIndex={0}
        type="text___blue"
        wrapperClassName={null}
      >
        Filters
      </Button>
    </div>
    <div
      className="Search_upper_right"
    >
      <IconButton
        className="Search_button___small"
        colorName="blue"
        data-test="closeFiltersButton"
        disabled={false}
        icon="tune"
        iconClassName={null}
        id="filtersButton"
        isLocked={false}
        kind="button"
        onClick={[Function]}
        style={null}
        tabIndex={0}
        type="outline___md"
      >
        Filters
      </IconButton>
      <ControlledInput
        autoComplete={null}
        className="Search_search"
        data-test="searchFilterInput"
        disabled={false}
        error={null}
        helperText={null}
        icon="search"
        id={null}
        kind="text"
        label={null}
        max={null}
        maxLength={null}
        min={null}
        name="search"
        onChange={[Function]}
        placeholder="Search"
        step={null}
        success={null}
        tabIndex={null}
        type="slim"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`RENDERING - Search should render the search filter if it has a search filter 1`] = `
<div
  className="Search___card"
>
  <div
    className="Search_upper"
  >
    <div
      className="Search_upper_left"
    />
    <div
      className="Search_upper_right"
    >
      <ControlledInput
        autoComplete={null}
        className="Search_search"
        data-test="searchFilterInput"
        disabled={false}
        error={null}
        helperText={null}
        icon="search"
        id={null}
        kind="text"
        label={null}
        max={null}
        maxLength={null}
        min={null}
        name="search"
        onChange={[Function]}
        placeholder="Search"
        step={null}
        success={null}
        tabIndex={null}
        type="slim"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`RENDERING - Search should render the sort options if there is/are sort option/s 1`] = `
<div
  className="Search___card"
>
  <div
    className="Search_upper"
  >
    <div
      className="Search_upper_left"
    />
    <div
      className="Search_upper_right"
    >
      <ControlledInput
        autoComplete={null}
        className="Search_search"
        data-test="searchFilterInput"
        disabled={false}
        error={null}
        helperText={null}
        icon="search"
        id={null}
        kind="text"
        label={null}
        max={null}
        maxLength={null}
        min={null}
        name="search"
        onChange={[Function]}
        placeholder="Search"
        step={null}
        success={null}
        tabIndex={null}
        type="slim"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`RENDERING - Search should render the sort options if there is/are sort option/s 2`] = `
<div
  className="Search___card"
>
  <div
    className="Search_upper"
  >
    <div
      className="Search_upper_left"
    >
      <div
        className="Search_group"
      >
        <Text
          className="Search_label"
          colorClass="n___400"
          element={null}
          id={null}
          type="body___md"
        >
          Sort by:
        </Text>
        <Connect(ControlledDropdown)
          className="Search_upper_dropdown"
          data-test="sortDropdown"
          name="sort"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Dummy Label",
                "value": 1,
              },
            ]
          }
          type="slim"
        />
      </div>
    </div>
    <div
      className="Search_upper_right"
    >
      <ControlledInput
        autoComplete={null}
        className="Search_search"
        data-test="searchFilterInput"
        disabled={false}
        error={null}
        helperText={null}
        icon="search"
        id={null}
        kind="text"
        label={null}
        max={null}
        maxLength={null}
        min={null}
        name="search"
        onChange={[Function]}
        placeholder="Search"
        step={null}
        success={null}
        tabIndex={null}
        type="slim"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`RENDERING - Search should render with default props 1`] = `
<div
  className="Search___card"
>
  <div
    className="Search_upper"
  >
    <div
      className="Search_upper_left"
    />
    <div
      className="Search_upper_right"
    >
      <ControlledInput
        autoComplete={null}
        className="Search_search"
        data-test="searchFilterInput"
        disabled={false}
        error={null}
        helperText={null}
        icon="search"
        id={null}
        kind="text"
        label={null}
        max={null}
        maxLength={null}
        min={null}
        name="search"
        onChange={[Function]}
        placeholder="Search"
        step={null}
        success={null}
        tabIndex={null}
        type="slim"
        value=""
      />
    </div>
  </div>
</div>
`;
